protocol/x/masa/masa.py

```diff
@@ -0,0 +1,224 @@
+        
+        # Override BASE_URL if provided
+        if base_url:
+            logger.debug(f"Overriding BASE_URL with: {base_url}")
+            if 'twitter' not in self.config:
+                self.config['twitter'] = {}
+            self.config['twitter']['BASE_URL'] = base_url
+            
+        # Initialize MasaSDK only once
+        self.masa_sdk = self._initialize_masa()
+        
+        # Set data directory
+        self.data_directory = data_directory or os.path.join(os.path.dirname(__file__), 'data')
+        os.makedirs(self.data_directory, exist_ok=True)
+        
+        # Configure logging
+        self.logger = logging.getLogger(__name__)
+
+    def _load_config(self, config_path: Optional[str] = None) -> Dict:
+        """Load configuration from YAML file"""
+        if not config_path:
+            config_path = Path(__file__).parent.parent.parent / "configs" / "x"
+        
+        settings_path = Path(config_path) / "settings.yaml"
+        logger.debug(f"Loading configuration from: {settings_path}")
+        
+        try:
+            with open(settings_path) as f:
+                config = yaml.safe_load(f)
+            config_data = config.get('default', {})
+            logger.debug(f"Loaded configuration: {config_data}")
+            return config_data
+        except Exception as e:
+            logger.error(f"Failed to load configuration: {e}")
+            raise
+    
+    def _initialize_masa(self) -> MasaSDK:
+        """Initialize Masa SDK client"""
+        try:
+            logger.debug("Initializing MasaSDK with default settings")
+            masa = MasaSDK()
+            
+            # Update settings through global_settings
+            logger.debug("Applying custom settings:")
+            for key, value in self.config.items():
+                if isinstance(value, dict):
+                    for subkey, subvalue in value.items():
+                        setting_key = f"{key}.{subkey}"
+                        logger.debug(f"Setting {setting_key} = {subvalue}")
+                        masa.global_settings[setting_key] = subvalue
+                else:
+                    logger.debug(f"Setting {key} = {value}")
+                    masa.global_settings[key] = value
+                    
+            return masa
+        except Exception as e:
+            logger.error(f"Failed to initialize Masa client: {e}")
+            raise
+            
+    def process_requests(self, requests: List[Dict[str, Any]]) -> Dict[str, Any]:
+        """
+        Process a list of requests
+        """
+        self.logger.info(f"Processing {len(requests)} requests")
+        
+        try:
+            # Initialize request processing
+            processed_requests = []
+            for request in requests:
+                # Format request according to MasaSDK expectations
+                formatted_request = {
+                    "type": request["scraper"],
```

Comment:
for now we can infer the response type based on the scraper type, but perhaps this will not always be the case?  are they meant to always be 1:1?